### ISSUE TITLE
关于生成的数据

### ISSUE THREAD

**Issue by holydick99** [Y-debug-sys/Diffusion-TS#54]
问题如下：
- 我传入的数据为（n，16）
- 当前输出为（2001，24，16），应该如何调整生成为（n，16）形状的形式

**Comment by holydick99** [comment]
> 问题如下：
> 
> * 我传入的数据为（n，16）
> * 当前输出为（2001，24，16），应该如何调整生成为（n，16）形状的形式

model:
  target: Models.interpretable_diffusion.gaussian_diffusion.Diffusion_TS
  params:
    seq_length: 24
    feature_size: 16
    n_layer_enc: 4
    n_layer_dec: 4
    d_model: 64  # 4 X 16
    timesteps: 256
    sampling_timesteps: 256
    loss_type: 'l1'
    beta_schedule: 'cosine'
    n_heads: 4
    mlp_hidden_times: 4
    attn_pd: 0.0
    resid_pd: 0.0
    kernel_size: 5
    padding_size: 2

solver:
  base_lr: 1.0e-5
  max_epochs: 10
  results_folder: ./Checkpoints_EEG
  gradient_accumulate_every: 2
  save_cycle: 1  # max_epochs // 10
  ema:
    decay: 0.995
    update_interval: 10
  
  scheduler:
    target: engine.lr_sch.ReduceLROnPlateauWithWarmup
    params:
      factor: 0.5
      patience: 3000
      min_lr: 1.0e-5
      threshold: 1.0e-1
      threshold_mode: rel
      warmup_lr: 8.0e-4
      warmup: 500 
      verbose: False

dataloader:
  train_dataset:
    target: Utils.Data_utils.real_datasets.EEGDataset
    params:
      name: eeg
      proportion: 1  # Set to rate < 1 if training conditional generation
      data_root: ./Data/datasets/EEG
      window: 24  # seq_length
      save2npy: True
      neg_one_to_one: True
      seed: 123
      period: train

  test_dataset:
    target: Utils.Data_utils.real_datasets.EEGDataset
    params:
      name: eeg
      proportion: 0.9  # rate
      data_root: ./Data/datasets/EEG
      window: 24  # seq_length
      save2npy: True
      neg_one_to_one: True
      seed: 123
      period: test
      style: separate
      distribution: geometric
    coefficient: 1.0e-2
    step_size: 5.0e-2
    sampling_steps: 200

  batch_size: 64
  sample_size: 256
  shuffle: True
  这是我的配置文件

**Comment by Y-debug-sys** [comment]
直接修改config文件里 出现24的地方，改为n，但是n太大的话会严重影响取样速度。

**Comment by holydick99** [comment]
> 直接修改config文件里 出现24的地方，改为n，但是n太大的话会严重影响取样速度。

奇怪，前面的2001是怎么出现的？我的配置文件里面并没有出现2001，max-epoch = 10 × 采样步数 = 200？

**Comment by Y-debug-sys** [comment]
定义的最小生成单位是2001，生成了2001个假数据的意思。

**Comment by holydick99** [comment]
> > 直接修改config文件里 出现24的地方，改为n，但是n太大的话会严重影响取样速度。
> 
> 奇怪，前面的2001是怎么出现的？我的配置文件里面并没有出现2001，max-epoch = 10 × 采样步数 = 200？

emm，那这个24是看作通道数？有点没有理解这个生成数据维度的含义，可以简单说明一下吗

**Comment by holydick99** [comment]
> 定义的最小生成单位是2001，生成了2001个假数据的意思。

我是这样理解生成的数据维度的，我传入的是一个（n，m）数据，然后模型最少生成2001个（n，m）的数据，是这样的吗？

**Comment by Y-debug-sys** [comment]
是的，可以调整，在 main.py 中相应位置。
